### ISSUE TITLE
Possibility to make users inactive (without deleting them)

### ISSUE THREAD

**Issue by Langveisfra** [meeting-room-booking-system/mrbs-code#3505]
If a user quits the organization, it should be possible for admin to set the user _inactive_.
Users with status _inactive_ should not be able to log in, and be excluded in the "created by"-list (in the entry-pages).
The inactive users should also be hidden when entering the edit_users view. It should instead be a checkbox for "showing inactive users".

The solution in version 1.11.2 and 1.11.3 (and possibly pervious versions) is to delete the user (and his/hers record) in the database.
When deleting a user you cripple the history. (The user info is blanked out from any reservation, future and past).

If there still a possibility to delete a user is a desired function, it should only be possible with users that doesn't have any records with entries in the "Created by"-field.

**Comment by campbell-m** [comment]
You could set the rights of the user to "none".  That would prevent them logging in without crippling history.   Would that achieve your goal?

**Comment by Langveisfra** [comment]
Yes, if the "none"-user does not have privileges to log on that is fine. 
In my opinion, _inactive_ is better terminology than _none_.
On the other hand, we will have 5 - 7 administrators. For safety reasons noe one of us should be allowed to delete users that are present in the booking data.

**Comment by campbell-m** [comment]
> Yes, if the "none"-user does not have privileges to log on that is fine.

They are able to log on, but will only be able to see pages that are open to the public, so in effect that's the same.

> In my opinion, inactive is better terminology than none.

You can always override the standard terminology in your config file:

~~~php
$vocab_override["en"]["level_0"]            = "inactive";
$vocab_override["sv"]["level_0"]            = "inaktiv";
~~~

> On the other hand, we will have 5 - 7 administrators. For safety reasons noe one of us should be allowed to delete users that are present in the booking data.

For some sites it will be useful to de able to delete users, even if they have bookings in the system.  The booking will still show their username, but not their display name.   How about (1) a warning that a user has bookings in the system and (2) a config option to disable deletion of users with bookings in the system, even with a warning?

**Comment by Langveisfra** [comment]
All suggestions = 👍👍👍

**Comment by Langveisfra** [comment]
Great!
